### DISCUSSION TITLE
Distance to trajectory

### DISCUSSION THREAD

**Discussion by jblich870** [GreenleafLab/ArchR#1668] [Questions / Documentation]
Hi,

Is it possible to extract a value for each cell of shortest distance to a given trajectory? In the Trajectory Analysis part of the tutorial it states that "ArchR aligns all cells to the trajectory based on their Euclidean distance to the nearest point along the manifold". I wondered whether this distance was stored or able to be retrieved. I would like to use this to compute cell weights for each trajectory in my dataset (similar to output by Slingshot) to input into Tradeseq.

Many thanks,
J

**Comment by rcorces** [comment]
This isnt currently stored anywhere but it wouldnt be hard to implement. Will try to do so this week. In the meantime, you could modify the `addTrajectory()` function to store the object `dfTrajectory3` somewhere. Thats where you will find the distance.

https://github.com/GreenleafLab/ArchR/blob/b7a2e8f683eba4d37f4eceaca514293f3b5a3db1/R/Trajectory.R#L238-L251

**Reply by jblich870** [reply]
Many thanks for the reply. Modifying the function is slightly beyond me, but implementing this feature would be great and I think a really useful extension of the function.
Thank you!
J

**Reply by rcorces** [reply]
this has been implemented in the `dev_trajectory` branch. Please confirm that this works for you and provides what you were hoping for and I will merge it into `dev` for eventual inclusion in `release_1.0.3`

**Reply by rcorces** [reply]
```
devtools::install_github("GreenleafLab/ArchR", ref="dev_trajectory", repos = BiocManager::repositories())
#to unload a package and reload
detach("package:ArchR", unload=TRUE)
library(ArchR)
```

**Reply by jblich870** [reply]
I see the `saveDF` parameter - where does it get stored?

**Reply by rcorces** [reply]
https://github.com/GreenleafLab/ArchR/blob/c8b9071fd0ef9d6355ee2c29eda2076e67885baf/R/Trajectory.R#L26-L27

**Reply by jblich870** [reply]
Ah yes, thanks. This is perfect, thank you for attending to this so quickly!

**Reply by rcorces** [reply]
is that confirmation that it works on your end?

**Reply by jblich870** [reply]
Yes, works well this end. Thank you!

**Reply by rcorces** [reply]
Merged to `dev` via https://github.com/GreenleafLab/ArchR/pull/1673
